pkg/auth/providers/saml/saml_handlers.go

```diff
@@ -31,7 +31,8 @@ func (s *Provider) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 				log.Debugf("RESPONSE: ===\n%s\n===\nNOW: %s\nERROR: %s",
 					parseErr.Response, parseErr.Now, parseErr.PrivateErr)
 			}
-			http.Error(w, http.StatusText(http.StatusForbidden), http.StatusForbidden)
+			redirectURL := r.URL.Host + "/login"
```

Comment:
Why redirect?

Reply:
If do not redirect, it will return an invalid page if we direct access a route path of authentication, I do not think this is a good UX, because we can access Rancher server from other(okta) platform.

Reply:
The redirect URL can also include the error code as a query parameter, because UI has logic to handle certain error codes. So can we append the error code here, like in line
https://github.com/rancher/rancher/blob/master/pkg/auth/providers/saml/saml_client.go#L279

Reply:
Awesome, good idea,  thank you!

Reply:
@mrajashree I've updated as you suggested, please review again.

Reply:
Thanks!